examples/connectivity/plot_extract_regions_canica_maps.py

```diff
@@ -0,0 +1,153 @@
+                         title=title)
+
+################################################################################
+# Computing correlation coefficients
+# First we need to do subjects timeseries signals extraction and then estimating
+# correlation matrices on those signals.
+# To extract timeseries signals, we call transform() from RegionExtractor object
+# onto each subject functional data stored in func_filenames.
+# To estimate correlation matrices we import connectome utilities from nilearn
+from nilearn.connectome import ConnectivityMeasure, sym_to_vec
+
+correlations = []
+# Initializing ConnectivityMeasure object with kind='correlation'
+connectome_measure = ConnectivityMeasure(kind='correlation')
+for filename, confound in zip(func_filenames, confounds):
+    # call transform from RegionExtractor object to extract timeseries signals
+    timeseries_each_subject = extractor.transform(filename, confounds=confound)
+    # call fit_transform from ConnectivityMeasure object
+    correlation = connectome_measure.fit_transform([timeseries_each_subject])
+    # saving each subject correlation to correlations
+    correlations.append(correlation)
+
+# Mean of all correlations
+import numpy as np
+
+mean_correlations = np.mean(correlations, axis=0).reshape(n_regions_extracted,
+                                                          n_regions_extracted)
+
+# Visualization
+# Showing mean correlation results
+# Import image utilities in utilising to operate on 4th dimension
+import numpy as np
+import matplotlib.pyplot as plt
+from nilearn import image
+
+regions_imgs = image.iter_img(regions_extracted_img)
+coords_connectome = [plotting.find_xyz_cut_coords(img) for img in regions_imgs]
+title = 'Correlation interactions between %d regions' % n_regions_extracted
+plt.figure()
+plt.imshow(mean_correlations, interpolation="nearest",
+           vmax=1, vmin=-1, cmap=plt.cm.bwr)
+plt.colorbar()
+plt.title(title)
+plotting.plot_connectome(mean_correlations, coords_connectome,
+                         edge_threshold='90%', title=title)
+
+################################################################################
+# Showing only Default Mode Network (DMN) regions before and after region extraction
+# by manually identifying the index of DMN in ICA decomposed components
+
+# First we plot DMN without region extraction, interested in only index=[3]
+img = image.index_img(components_img, 3)
+coords = plotting.find_xyz_cut_coords(img)
+display = plotting.plot_anat(cut_coords=((0, -52, 29)), title='ICA map: DMN')
+display.add_contours(img, levels=[0.007], linewidths=2.5, colors=[[0., 0.092711, 1., 1.]],
+                     filled=True, linestyles='solid', alpha=0.5)
+
+# Now, we plot DMN after region extraction to show that connected regions are
+# nicely separated. Each brain extracted region is indicated with separate color
+extracted_DMN_indices = [10, 14, 15, 17]
```

Comment:
Maybe it would be better if you retrieved these indices from the region_index list.

And then you could do a for loop over them below.

The reason I am asking this is that I have the impression that not all regions of the original map are shown.